src/java/org/jivesoftware/openfire/archive/ConversationManager.java

```diff
@@ -130,6 +130,69 @@
     private Archiver<ArchivedMessage> messageArchiver;
     private Archiver<RoomParticipant> participantArchiver;
 
+    public static SystemProperty<Boolean> METADATA_ARCHIVING_ENABLED = SystemProperty.Builder.ofType(Boolean.class)
+        .setKey("conversation.metadataArchiving")
+        .setDefaultValue(true)
+        .setDynamic(true)
+        .setPlugin("monitoring")
+        .build();
+
+    public static SystemProperty<Boolean> MESSAGE_ARCHIVING_ENABLED = SystemProperty.Builder.ofType(Boolean.class)
+        .setKey("conversation.messageArchiving")
+        .setDefaultValue(false)
+        .setDynamic(true)
+        .setPlugin("monitoring")
+        .build();
+
+    public static SystemProperty<Boolean> ROOM_ARCHIVING_ENABLED = SystemProperty.Builder.ofType(Boolean.class)
+        .setKey("conversation.roomArchiving")
+        .setDefaultValue(false)
+        .setDynamic(true)
+        .setPlugin("monitoring")
+        .build();
+
+    public static SystemProperty<Boolean> ROOM_STANZA_ARCHIVING_ENABLED = SystemProperty.Builder.ofType(Boolean.class)
+        .setKey("conversation.roomArchivingStanzas")
+        .setDefaultValue(false)
+        .setDynamic(true)
+        .setPlugin("monitoring")
+        .build();
+
+    public static SystemProperty<String> ROOMS_ARCHIVED = SystemProperty.Builder.ofType(String.class)
+        .setKey("conversation.roomsArchived")
+        .setDefaultValue("")
+        .setDynamic(true)
+        .setPlugin("monitoring")
+        .build();
+
+    public static SystemProperty<Integer> IDLE_TIME = SystemProperty.Builder.ofType(Integer.class)
+        .setKey("conversation.idleTime")
+        .setDefaultValue(DEFAULT_IDLE_TIME)
+        .setDynamic(true)
+        .setPlugin("monitoring")
+        .build();
+
+    public static SystemProperty<Integer> MAX_TIME = SystemProperty.Builder.ofType(Integer.class)
```

Comment:
This would display better on the console if it were a Duration (saved as ChronoUnit.DAYS) instead of a n integer